druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupOperatorConversion.java

```diff
@@ -66,9 +67,15 @@ public DruidExpression toDruidExpression(PlannerContext plannerContext, RowSigna
                     final Expr columnName = inputExpressions.get(2).parse(plannerContext.getExprMacroTable()); // maha lookup name
 
                     LookupReferencesManager lrm = (LookupReferencesManager) lookupReferencesManager;
-                    String missingValue = getMissingValue(inputExpressions, plannerContext);
-
-                    if (arg.isSimpleExtraction() && lookupName.isLiteral() && columnName.isLiteral()) {
+                    String missingValue = getMissingValue(inputExpressions, plannerContext, 3, MISSING_VALUE);
+                    String extractionMap = getMissingValue(inputExpressions, plannerContext, 4, "-3");
+                    String valueExtractionMap = getMissingValue(inputExpressions, plannerContext, 5, "-3");
+                    //TODO: Enhance by passing in KV separator & delimeter.
+                    //Also, allow passing of Map type statements: Case, KV pair, etc. ex: CASE, MAP(',','->')
+                    Map<String, String> dimensionOverrideMap = extractionMap == null || extractionMap.equals("-3") ? null : Splitter.on(",").withKeyValueSeparator("->").split(extractionMap);
```

Comment:
Good that we are using Guava for string to map. Makes life easy :) 
We can move separator and map separator to static constant value in this class.